raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -941,6 +949,8 @@ private FetchResponseData tryCompleteFetchRequest(
     ) {
         Optional<Errors> errorOpt = validateLeaderOnlyRequest(request.currentLeaderEpoch());
         if (errorOpt.isPresent()) {
+            // TODO: The replica should return what information it knows about the current epoch and
```

Comment:
Isn't this already done by `buildFetchResponse`?

Reply:
Yes. This was a note to me to confirm this as it wasn't clear to me when I first read the code.